monkey/monkey_island/cc/ui/src/services/AuthService.js

```diff
@@ -29,16 +33,51 @@ export default class AuthService {
       .then(response => response.json())
       .then(response => {
         if(response.meta.code === 200){
-          this._removeToken();
+          this._removeAuthToken();
+          this._removeAuthTokenExpirationTime();
         }
         return response;
       });
   }
 
-  authFetch = (url, options) => {
+  authFetch = (url, options, refreshToken = false) => {
+    // refreshToken is a mechanism to keep unneeded calls
+    // to the refresh authentication token endpoint
+    if(this._shouldRefreshToken() && refreshToken){
+      this._refreshAuthToken();
+    }
     return this._authFetch(url, options);
   };
```

Comment:
Do we have to worry about concurrency issues? Is it possible tor two threads to call this simultaneously?

Reply:
I don't think that anything is calling `authFetch` concurrently